src/Core/Domain/Store/Command/AddStoreCommand.php

```diff
@@ -0,0 +1,383 @@
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Store\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+use PrestaShop\PrestaShop\Core\Domain\Store\ValueObject\Coordinate;
+use PrestaShop\PrestaShop\Core\Domain\Store\ValueObject\Hours;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Address;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\CityName;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Email;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\GenericName;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Note;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\PhoneNumber;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Postcode;
+
+/**
+ * Class AddStoreCommand is responsible for adding store data.
+ */
+class AddStoreCommand
+{
+    /**
+     * @var CountryId
+     */
+    private $countryId;
+
+    /**
+     * @var StateId|null
+     */
+    private $stateId;
+
+    /**
+     * @var Postcode
+     */
+    private $postcode;
+
+    /**
+     * @var CityName
+     */
+    private $city;
+
+    /**
+     * @var Coordinate|null
+     */
+    private $latitude;
+
+    /**
+     * @var Coordinate|null
+     */
+    private $longitude;
+
+    /**
+     * @var PhoneNumber|null
+     */
+    private $phone;
+
+    /**
+     * @var PhoneNumber|null
+     */
+    private $fax;
+
+    /**
+     * @var Email|null
+     */
+    private $email;
+
+    /**
+     * @var int[]
+     */
+    private $shopAssociation;
+
+    /**
+     * @var array<int, GenericName>
+     */
+    private $localizedNames = [];
+
+    /**
+     * @var array<int, Address>
+     */
+    private $localizedAddress1 = [];
+
+    /**
+     * @var array<int, Address>
+     */
+    private $localizedAddress2 = [];
+
+    /**
+     * @var array<int, Hours>
+     */
+    private $localizedHours = [];
+
+    /**
+     * @var array<int, Note>
+     */
+    private $localizedNotes = [];
+
+    /**
+     * @param int[] $shopAssociation
```

Comment:
Please update PHPDoc

Reply:
PHPDoc on top of typehints? Isn't it overkill?

Reply:
Yes and No (because old IDE doesn't detect the typehint).

Reply:
I agree its overkill when all types are clearly defined, except with arrays. You don't see what is the type of array elements, so it is definitely useful to see psalm annotations in such cases (also the phpstan complains when those types mismatch :v:  )